### PR TITLE
feat(driver-adapters): enable Wasm on `sql-query-connector`

### PR DESCRIPTION
**Pull request by jkomyno**
This PR closes https://github.com/prisma/team-orm/issues/283, and depends on https://github.com/prisma/prisma-engines/pull/4442.

### CODE REVIEW

File: query-engine/connectors/sql-query-connector/src/database/mod.rs [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
-pub use postgresql::*;
-pub use sqlite::*;
+
+#[cfg(not(target_arch = "wasm32"))]
+pub use native::{mssql::*, mysql::*, postgresql::*, sqlite::*};
+
+// pub use mssql::*;
+// pub use mysql::*;
+// pub use postgresql::*;
+// pub use sqlite::*;
```
Comment (jkomyno) [inline query-engine/connectors/sql-query-connector/src/database/mod.rs:28]:
```suggestion
```

File: query-engine/connectors/sql-query-connector/src/database/operations/write.rs [hunk 2]
```diff
@@ -18,9 +18,28 @@ use std::{
...
 
+#[cfg(target_arch = "wasm32")]
+macro_rules! trace {
+    (target: $target:expr, $($arg:tt)+) => {{
+        // No-op in WebAssembly
+    }};
+    ($($arg:tt)+) => {{
+        // No-op in WebAssembly
+    }};
+}
```
Comment (jkomyno) [inline query-engine/connectors/sql-query-connector/src/database/operations/write.rs:31]: `trace!` macro for `wasm32-*`

File: query-engine/connectors/sql-query-connector/src/database/operations/write.rs [hunk 3]
```diff
@@ -18,9 +18,28 @@ use std::{
...
+
+#[cfg(not(target_arch = "wasm32"))]
+macro_rules! trace {
+    (target: $target:expr, $($arg:tt)+) => {
+        tracing::log::trace!(target: $target, $($arg)+);
+    };
+    ($($arg:tt)+) => {
+        tracing::log::trace!($($arg)+);
+    };
+}
```
Comment (jkomyno) [inline query-engine/connectors/sql-query-connector/src/database/operations/write.rs:41]: `trace!` macro for native compilation.

**Review by miguelff — APPROVED** [review]
Changes look correct and reasonably tidy.

File: query-engine/connectors/sql-query-connector/Cargo.toml [hunk 4]
```diff
@@ -18,13 +20,13 @@ once_cell = "1.3"
...
 thiserror = "1.0"
-tokio.workspace = true
+tokio = { version = "1.0", features = ["macros", "time"] }
 tracing = "0.1"
 tracing-futures = "0.2"
 uuid.workspace = true
 opentelemetry = { version = "0.17", features = ["tokio"] }
 tracing-opentelemetry = "0.17.3"
-quaint.workspace = true
+quaint = { path = "../../../quaint" }
```
[the remark this answers is not in the document]
Comment (jkomyno) [reply query-engine/connectors/sql-query-connector/Cargo.toml:29]: Yes, but I spotted a problem while running the CI for `request-handlers`.
I'm cherry-picking https://github.com/prisma/prisma-engines/pull/4455/commits/96cd8ca800176eaeb9cf803c2b7e340973eaddc0 here.

[the remark this answers is not in the document]
Comment (jkomyno) [reply query-engine/connectors/sql-query-connector/Cargo.toml]: (This indeed fixed the CI tests)

**Pull request merged**